watertap/examples/flowsheets/case_studies/groundwater_treatment/groundwater_treatment.py

```diff
@@ -0,0 +1,346 @@
+
+    m.fs.feed.flow_vol[0].fix(flow_vol)
+    m.fs.feed.conc_mass_comp[0, "arsenic"].fix(conc_mass_arsenic)
+    m.fs.feed.conc_mass_comp[0, "uranium"].fix(conc_mass_uranium)
+    m.fs.feed.conc_mass_comp[0, "nitrate"].fix(conc_mass_nitrate)
+    m.fs.feed.conc_mass_comp[0, "phosphates"].fix(conc_mass_phosphates)
+    m.fs.feed.conc_mass_comp[0, "iron"].fix(conc_mass_iron)
+    m.fs.feed.conc_mass_comp[0, "filtration_media"].fix(conc_mass_filtration_media)
+
+    solve(m.fs.feed)
+
+    # pump
+    m.fs.pump.load_parameters_from_database(use_default_removal=True)
+    m.fs.pump.lift_height.fix(30)  # head of 30 m to pump groundwater
+
+    # microbial battery
+    m.fs.micbatt.load_parameters_from_database(use_default_removal=True)
+
+
+def initialize_system(m):
+    seq = SequentialDecomposition()
+    seq.options.tear_set = []
+    seq.options.iterLim = 1
+    seq.run(m, lambda u: u.initialize())
+
+
+def solve(blk, solver=None, tee=False, check_termination=True):
+    if solver is None:
+        solver = get_solver()
+    results = solver.solve(blk, tee=tee)
+    if check_termination:
+        assert_optimal_termination(results)
+    return results
+
+
+def display_results(m):
+    print("\n+++++++++++++++++++++++++++++++++++++++++++++++++++++++++")
+    print("++++++++++++++++++++ DISPLAY RESULTS ++++++++++++++++++++")
+    print("+++++++++++++++++++++++++++++++++++++++++++++++++++++++++")
+
+    unit_list = ["feed", "pump", "micbatt"]
+    for u in unit_list:
+        m.fs.component(u).report()
+
+    print("\n+++++++++++++++++++++++++++++++++++++++++++++++++++++++++")
+
+
+def add_costing(m):
+    source_file = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)),
+        "groundwater_treatment_case_study.yaml",
+    )
+
+    m.fs.costing = ZeroOrderCosting(default={"case_study_definition": source_file})
+
+    costing_kwargs = {"default": {"flowsheet_costing_block": m.fs.costing}}
+    m.fs.pump.costing = UnitModelCostingBlock(**costing_kwargs)
+    m.fs.micbatt.costing = UnitModelCostingBlock(**costing_kwargs)
+
+    m.fs.costing.cost_process()
+
+    m.fs.costing.add_electricity_intensity(m.fs.feed.properties[0].flow_vol)
+
+    m.fs.costing.spent_media_recovery_mass = Expression(
+        expr=m.fs.costing.utilization_factor
+        * pyunits.convert(
+            m.fs.byproduct.flow_mass_comp[0, "filtration_media"],
+            to_units=pyunits.kg / m.fs.costing.base_period,
+        ),
+        doc="Mass of spent filtration media generated per year (accounting for utilization factor)",
+    )
+
+    m.fs.costing.cost_spent_media_disposal = Expression(
+        expr=pyunits.convert(
+            m.fs.costing.spent_media_recovery_mass
+            * m.fs.costing.filtration_media_disposal_cost,
+            to_units=m.fs.costing.base_currency / m.fs.costing.base_period,
+        ),
+        doc="Disposal cost of spent filtration media per year (accounting for utilization factor)",
```

Comment:
I am thinking we should consider a variable for recovering some fraction of the "spent_media," assuming regeneration of media and reuse. Right now, it seems that the model doesn't consider any regeneration. We may have talked about how we were going to handle this, but I think I need a refresher. For the time being, it seems like the current implementations accounts for the worst-case scenario (no regeneration+recovery of media) and doesn't leave room for the potential case where media could be recovered and reused.

Reply:
Given that we are still talking about zero-order models at this point, I would suggest that this would be best handled in an abstract manner. I.e. the new media stream would only include the new, fresh media required (minus an recycling), and that there should be a separate operating cost for media regeneration included in the unit operation (and incorporate any capital costs for regeneration into the unit capital cost) Thus, the flowsheet would not see an of the media regeneration at all.

Reply:
Without some recycle/regeneration of the media, it would technically leave the system and would be accounted for in sludge disposal costs, but that is not what was originally intended by the group-- they intended to account for regen/recycle. I'm trying to recall the original intent by @tarnold17 here, and I think it had to do partially with the fact that iron is the key reactant and is oxidized to form the filtration media. So, filtration media takes part in the reaction in this example (yet, iron only needs to be seeded into the system for media generation to commence).

I did suggest that we could do without including the media as a component and just consider related capex and opex as an alternative. At this point though, we only intend to put together some initial cost results and reiterate with the team on our implementation and results.

Reply:
> I think it had to do partially with the fact that iron is the key reactant and is oxidized to form the filtration media

This is precisely why I included filtration media as one of the components - it is essentially a product of one of the reactions that occurs in the unit. I do agree that there could have been other ways to model this; perhaps there is another approach that would have been better. We can certainly refine if needed.